R/plot_fundamental_diagram.R

```diff
@@ -0,0 +1,328 @@
+    tolower(weekday_choice)
+    enriched_data<-enriched_data %>% filter(.data$weekday %in% weekday_choice)
+  }
+
+  if(!is.null(hour_choice))
+  {enriched_data<-enriched_data %>% filter(.data$hour %in% hour_choice)}
+
+  if(!is.null(vacation_choice))
+  {enriched_data<-enriched_data %>% filter(.data$vacation %in% vacation_choice)}
+
+  if(!is.null(holiday_choice))
+  {enriched_data<-enriched_data %>% filter(.data$holiday %in% holiday_choice)}
+
+  enriched_data$Version<-ifelse(enriched_data$segment_id %in% version_capteur[10:length(version_capteur$Id_Segment),1],2,1)
+
+  enriched_data$weekend<-ifelse(enriched_data$weekday %in% c('saturday','sunday'), "Week-end", "Semaine")
+  return(enriched_data)
+}
+
+#' Calculate necessary data to trace the fundamental diagram
+#'
+#' @param enriched_data enriched data.frame containing all the data for all your sensors
+#' @param sensor_version Integer (1,2). Selected the version of the sensor (data from Telraam). Default to 1.
+#' @param direction_choice Character vector. Selected direction ('rgt','lft'). Both by default (NULL)
+#'
+#' @return enriched_data
+#' @export
+#'
+#'
+#' @examples
+#' calculate_axes(traffic)
+#' calculate_axes(traffic,
+#'   sensor_version=2,
+#'   direction_choice='rgt')
+calculate_axes <- function(enriched_data, sensor_version = 1, direction_choice=NULL)
+{
+  if(sensor_version==1)
+  {
+      enriched_data$veh_h<-enriched_data$car
+      enriched_data$km_h<-enriched_data$v85
+      enriched_data$veh_km<-0
+      for(i in 1:length(enriched_data$veh_km))
+        {enriched_data$veh_km[i]<-enriched_data$veh_h[i]/enriched_data$km_h[i]}
+  }
+
+  else
+  {
+    if(direction_choice=='lft')
+    {
+      enriched_data$veh_h_lft<-enriched_data$car_lft*4
+      enriched_data$km_h_lft<-enriched_data$v85_lft
+      enriched_data$veh_km_lft<-0
+      for(i in 1:length(enriched_data$car_lft))
+        {enriched_data$veh_km_lft[i]<-enriched_data$veh_h_lft[i]/enriched_data$km_h_lft[i]}
+    }
+
+    else
+    {
+      enriched_data$veh_h_rgt<-enriched_data$car_rgt*4
+      enriched_data$km_h_rgt<-enriched_data$v85_rgt
+      enriched_data$veh_km_rgt<-0
+      for(i in 1:length(enriched_data$car_rgt))
+        {enriched_data$veh_km_rgt[i]<-enriched_data$veh_h_rgt[i]/enriched_data$km_h_rgt[i]}
+    }
+  }
+  return(enriched_data)
+}
+
+#' Plot three fundamentals diagrams
+#'
+#' @param enriched_data enriched data.frame containing all the data for all your sensors
+#' @param sensor_version Integer (1,2). Selected the version of the sensor (data from Telraam). Default to 1.
+#' @param direction_choice Character vector. Selected direction ('rgt','lft'). Both by default (NULL)
+#'
+#' @return enriched_data
+#' @export
+#'
+#' @import ggplot2
+#'
+#' @examples
+#' plot_diagramm(traffic)
+#' plot_diagramm(traffic,
+#'   sensor_version=2,
+#'   direction_choice='rgt')
+plot_diagramm <- function(enriched_data, sensor_version = 1, direction_choice=NULL)
```

Comment:
there should be a way to generalize the code: you write the same thing three times. Maybe add a function with parameters that can be called within your conditions
+ no french (debit/vitesse/concentration -->  flow/speed/density )